scripts/lib/footnote.py

```diff
@@ -93,16 +92,37 @@ def form_footnote(
 def make(html: str) -> str:
     footnotes = {}
 
-    for fn_re in FOOTNOTES_RE:
-        html = fn_re["description"].sub(
+    i = 0
+    while len(footnotes) == 0 and i < len(FOOTNOTES_RE):
+        html = FOOTNOTES_RE[i]["description"].sub(
             lambda x: parse_description(x, footnotes),
             html,
         )
+        i += 1
+
+    if len(footnotes) == 0:
+        return html
+
+    html = FOOTNOTES_RE[i - 1]["link"].sub(
+        lambda x: form_footnote(x, footnotes),
+        html,
+    )
 
-        if len(footnotes) > 0:
-            return fn_re["link"].sub(
-                lambda x: form_footnote(x, footnotes),
-                html,
-            )
+    fn_without_link = list(
+        map(
+            lambda x: str(x),
+            filter(
+                lambda k: footnotes[k]["count"] == 0,
+                footnotes.keys(),
+            ),
+        )
+    )
```

Comment:
List comprehension will by more pythonc
```
fn_without_link = [str(x) for x in footnotes.keys() if footnotes[k]["count"] == 0]
```
and as it's immutable data it can be tuple instead of list